addon/attributes.js

```diff
@@ -295,9 +295,14 @@ function fragmentArrayProperty(metaType, options, createArray) {
       fragments = data;
     // Create a fragment array and initialize with data
     } else if (data && data !== fragments) {
-      fragments || (fragments = createArray(record, key));
-      internalModel._recordData._data[key] = fragments;
-      fragments.setupData(data);
+      if (fragments) {
+        internalModel._recordData._data[key] = fragments;
+        fragments.setupData(data);
+      } else {
+        fragments = createArray(record, key);
+        internalModel._recordData._data[key] = fragments;
+        fragments.initData(data);
```

Comment:
This is the fix, add a new method that can take advantage of the fact that we know content is empty and avoid reading from it.